tests/core/CoreTest.py

```diff
@@ -614,3 +614,15 @@ def test_run_heavy_cpu_load(self):
         self.assertEqual(len(result_set), len(results))
         self.assertEqual(result_set, {i for i in range(100)})
         self.assertEqual(bear.dependency_results, {})
+
+    def test_custom_executor(self):
+        bear = MultiTaskBear(self.section1, self.filedict1, tasks_count=100)
+
+        import concurrent.futures
+        results = self.execute_run(
+            {bear}, concurrent.futures.ThreadPoolExecutor(max_workers=10))
+
+        result_set = set(results)
+        self.assertEqual(len(result_set), len(results))
+        self.assertEqual(result_set, {i for i in range(100)})
```

Comment:
that was actually my bad for the tests I've written earlier: you can do `set(range(100))` :D